src/test/scala/com/microsoft/hyperspace/index/IcebergIntegrationTest.scala

```diff
@@ -0,0 +1,190 @@
+        val dfFromSample2 = sampleData2.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        dfFromSample2.write.format("iceberg").mode("append").save(dataPath)
+
+        // The index should not be applied for the updated version.
+        assert(!isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex"))
+
+        // The index should be applied for the version at index creation.
+        assert(isIndexUsed(
+          query(Some(currentIceSnapshot)).queryExecution.optimizedPlan, "iceIndex"))
+
+        hyperspace.refreshIndex("iceIndex")
+
+        // The index should be applied for the updated version.
+        assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex/v__=1"))
+
+        // The index should not be applied for the version at index creation.
+        assert(!isIndexUsed(query(Some(currentIceSnapshot)).queryExecution.optimizedPlan,
+          "iceIndex"))
+      }
+    }
+  }
+
+  test("Verify Hybrid Scan on Iceberg table.") {
+    withTempPathAsString { dataPath =>
+      import spark.implicits._
+      val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+
+      createIcebergTable(dataPath, dfFromSample)
+      dfFromSample.write.format("iceberg").mode("overwrite").save(dataPath)
+
+      val iceDf = spark.read.format("iceberg").load(dataPath)
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+        hyperspace.createIndex(iceDf, IndexConfig("iceIndex", Seq("clicks"), Seq("Query")))
+      }
+
+      withIndex("iceIndex") {
+        def query(version: Option[Long] = None): DataFrame = {
+          if (version.isDefined) {
+            val iceDf =
+              spark.read.format("iceberg").option("snapshot-id", version.get).load(dataPath)
+            iceDf.filter(iceDf("clicks") <= 2000).select(iceDf("query"))
+          } else {
+            val iceDf = spark.read.format("iceberg").load(dataPath)
+            iceDf.filter(iceDf("clicks") <= 2000).select(iceDf("query"))
+          }
+        }
+
+        assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex", false))
+
+        // Create a new version by adding entries.
+        val dfFromSample2 = sampleData2.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        dfFromSample2.write.format("iceberg").mode("append").save(dataPath)
+
+        withSQLConf(
+          "spark.hyperspace.index.hybridscan.enabled" -> "true",
+          "spark.hyperspace.index.hybridscan.delete.enabled" -> "true") {
+          // The index should be applied for the updated version.
+          assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex", true))
+
+          // Append data.
+          dfFromSample
+            .limit(3)
+            .write
+            .format("iceberg")
+            .mode("append")
+            .save(dataPath)
+
+          // The index should be applied for the updated version.
+          assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex", true))
```

Comment:
Could you share the result of `query().queryExecution.optimizedPlan/sparkPlan` here? This is the "appended files only" case in https://github.com/microsoft/hyperspace/pull/321#issuecomment-760099407 :)
I think we could add the result - transformed plan - in comments below until correct Hybrid Scan tests for Iceberg is delivered later.